lib/EarlyFunctionAnalysis/EarlyFunctionAnalysis.cpp

```diff
@@ -843,6 +875,105 @@ void FunctionEntrypointAnalyzer::interproceduralPropagation() {
   }
 }
 
+void FEA::initializeMapForDeductions(FunctionSummary &Summary,
+                                     abi::RegisterState::Map &Map) {
+  using namespace abi;
+  using namespace ABIAnalyses;
+  using RegisterState = abi::RegisterState::Values;
+  using Register = model::Register::Values;
+
+  auto GPRArguments = listGeneralPurposeArgumentRegisters(Binary->DefaultABI);
+  auto GPRRetVals = listGeneralPurposeReturnValueRegisters(Binary->DefaultABI);
+  auto VRArguments = listVectorArgumentRegisters(Binary->DefaultABI);
+  auto VRRetVals = listVectorReturnValueRegisters(Binary->DefaultABI);
+
+  auto TryGetRegisterState =
+    [this](Register Reg,
+           RegisterStateMap &ABIRegisterMap) -> std::optional<RegisterState> {
+    llvm::StringRef Name = model::Register::getCSVName(Reg);
+    if (llvm::GlobalVariable *CSV = M.getGlobalVariable(Name, true)) {
+      auto It = ABIRegisterMap.find(CSV);
+      if (It != ABIRegisterMap.end())
+        if (isYesOrDead(It->second))
+          return It->second;
+    }
+
+    return {};
+  };
+
+  for (auto &GPRArg : GPRArguments)
+    if (auto MaybeState = TryGetRegisterState(GPRArg,
+                                              Summary.ABIResults
+                                                .ArgumentsRegisters))
+      Map[GPRArg].IsUsedForPassingArguments = *MaybeState;
+
+  for (auto &GPRRetVal : GPRRetVals)
+    if (auto MaybeState = TryGetRegisterState(GPRRetVal,
+                                              Summary.ABIResults
+                                                .FinalReturnValuesRegisters))
+      Map[GPRRetVal].IsUsedForReturningValues = *MaybeState;
+
+  for (auto &VRArg : VRArguments)
+    if (auto MaybeState = TryGetRegisterState(VRArg,
+                                              Summary.ABIResults
+                                                .ArgumentsRegisters))
+      Map[VRArg].IsUsedForPassingArguments = *MaybeState;
+
+  for (auto &VRRetVal : VRRetVals)
+    if (auto MaybeState = TryGetRegisterState(VRRetVal,
+                                              Summary.ABIResults
+                                                .FinalReturnValuesRegisters))
+      Map[VRRetVal].IsUsedForReturningValues = *MaybeState;
```

Comment:
These lines starting from `.` feel pretty bad outside of method chaining idiom. And I kind of feel responsible for changes to the convention enforcing file (I'm referring to some old mattermost discussion).

Maybe something can done to make them a tad bit prettier? Like an alias or shortening the names of both `MaybeState` and `TryGetRegisterState`.